filters.go

```diff
@@ -274,3 +336,21 @@ func (p *Pipe) ReplaceRegexp(re *regexp.Regexp, replace string) *Pipe {
 		out.WriteRune('\n')
 	})
 }
+
+// TrimExt reads a list of filenames from the pipe, one per line, and returns
+// a pipe with the given file extension (e.g. '.txt') removed.
+//
+// File extensions are only removed if they match the one you provide.
+// Trailing dots '.' are only removed if you include it in your 'ext'.
+func (p *Pipe) TrimExt(ext string) *Pipe {
+	// do we have anything to do?
+	if p == nil || p.Error() != nil {
+		return p
+	}
+
+	return p.EachLine(func(line string, out *strings.Builder) {
+		newPath := strings.TrimSuffix(line, ext)
```

Comment:
I'm wondering if we shouldn't simply make this method `TrimSuffix`?

Reply:
I think it depends on your target audience. If you're aiming this at folks who are looking to port shell scripts to Golang, I'd stick with naming and behaviour that fits with their perspective. It'll feel more natural to them, because they won't have to work quite as hard to map things across.

Your library, your call :)

Reply:
Is there a Unix command called `trimext`? I've never come across it. I agree it's nice if things feel familiar to shell scripters, but at the same time I'm not willing to make UX sacrifices for this. After all, one reason for this library to exist is that programming in shell is such a miserable experience!

Reply:
File extension trimming is normally either done by `basename(1)` or using `x=${x#.ext}`. There isn't an explicit UNIX command for that.

Anyway, I've taken the feature out of this PR.